vpr/src/place/weighted_median_move_generator.cpp

```diff
@@ -0,0 +1,178 @@
+    }
+
+    if ((place_move_ctx.X_coord.size() == 0) || (place_move_ctx.Y_coord.size() == 0))
+        return e_create_move::ABORT;
+
+    //calculate the weighted median region
+    std::sort(place_move_ctx.X_coord.begin(), place_move_ctx.X_coord.end());
+    std::sort(place_move_ctx.Y_coord.begin(), place_move_ctx.Y_coord.end());
+
+    if (place_move_ctx.X_coord.size() == 1) {
+        limit_coords.xmin = place_move_ctx.X_coord[floor((place_move_ctx.X_coord.size() - 1) / 2)];
+        limit_coords.xmax = limit_coords.xmin;
+    } else {
+        limit_coords.xmin = place_move_ctx.X_coord[floor((place_move_ctx.X_coord.size() - 1) / 2)];
+        limit_coords.xmax = place_move_ctx.X_coord[floor((place_move_ctx.X_coord.size() - 1) / 2) + 1];
+    }
+
+    if (place_move_ctx.Y_coord.size() == 1) {
+        limit_coords.ymin = place_move_ctx.Y_coord[floor((place_move_ctx.Y_coord.size() - 1) / 2)];
+        limit_coords.ymax = limit_coords.ymin;
+    } else {
+        limit_coords.ymin = place_move_ctx.Y_coord[floor((place_move_ctx.Y_coord.size() - 1) / 2)];
+        limit_coords.ymax = place_move_ctx.Y_coord[floor((place_move_ctx.Y_coord.size() - 1) / 2) + 1];
+    }
+
+    t_range_limiters range_limiters;
+    range_limiters.original_rlim = rlim;
+    range_limiters.dm_rlim = placer_opts.place_dm_rlim;
+    range_limiters.first_rlim = place_move_ctx.first_rlim;
+
+    t_pl_loc w_median_point;
+    w_median_point.x = (limit_coords.xmin + limit_coords.xmax) / 2;
+    w_median_point.y = (limit_coords.ymin + limit_coords.ymax) / 2;
+    if (!find_to_loc_centroid(cluster_from_type, from, w_median_point, range_limiters, to)) {
+        return e_create_move::ABORT;
+    }
+    return ::create_move(blocks_affected, b_from, to);
+}
+
+/* This routine finds the bounding box of a net from scratch excluding   *
+ * a specific block. This is very useful in some directed moves.         *
+ * It updates coordinates of the bb                                      */
+static void get_bb_cost_for_net_excluding_block(ClusterNetId net_id, t_bb_cost* coords, ClusterBlockId, ClusterPinId moving_pin_id, const PlacerCriticalities* criticalities, bool& skip_net) {
+    int pnum, x, y, xmin, xmax, ymin, ymax;
+    float xmin_cost, xmax_cost, ymin_cost, ymax_cost, cost;
+
+    skip_net = true;
+
+    xmin = 0;
+    xmax = 0;
+    ymin = 0;
+    ymax = 0;
+    cost = 0.0;
+    xmin_cost = 0.0;
+    xmax_cost = 0.0;
+    ymin_cost = 0.0;
+    ymax_cost = 0.0;
+
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& place_ctx = g_vpr_ctx.placement();
+    auto& device_ctx = g_vpr_ctx.device();
+    auto& grid = device_ctx.grid;
+
+    ClusterBlockId bnum;
+    bool is_first_block = true;
+    int ipin;
+    for (auto pin_id : cluster_ctx.clb_nlist.net_pins(net_id)) {
+        bnum = cluster_ctx.clb_nlist.pin_block(pin_id);
+
+        //if(bnum != block_id)
```

Comment:
Delete the commented line if it is dead code.

Reply:
done